features/integration/simulate.feature

```diff
@@ -0,0 +1,84 @@
+Feature: Simulating transactions
+  Background:
+    Given an algod v2 client
+    And a kmd client
+    And wallet information
+    And suggested transaction parameters from the algod v2 client
+    And I create a new transient account and fund it with 100000000 microalgos.
+    And I make a transaction signer for the transient account.
+
+  @simulate
+  Scenario Outline: Simulating successful payment transactions
+    Given default transaction with parameters <amt> "<note>"
+    When I get the private key
+    And I sign the transaction with the private key
+    And I simulate the transaction
+    Then the simulation should succeed without any failure message
+
+    Examples:
+      | amt     | note         |
+      | 0       | X4Bl4wQ9rCo= |
+      | 1234523 | X4Bl4wQ9rCo= |
+
+  @simulate
+  Scenario: Simulating duplicate transactions in a group
+    Given a new AtomicTransactionComposer
+    When I build a payment transaction with sender "transient", receiver "transient", amount 100001, close remainder to ""
+    And I create a transaction with signer with the current transaction.
+    And I add the current transaction with signer to the composer.
+    Then I simulate the current transaction group with the composer
+    And the simulation should succeed without any failure message
+    And I clone the composer.
+    When I add the current transaction with signer to the composer.
+    Then I simulate the current transaction group with the composer
+    And the simulation should report a failure at path "1" with message "transaction already in ledger"
```

Comment:
I see the following nested response structure:
```go
Result{
		Version:      version,
		LastRound:    lastRound,
		TxnGroups:    []TxnGroupResult{
                               	Txns:           []TxnResult {
                               	                           	Txn:                        transactions.SignedTxnWithAD
                               	                           	MissingSignature: bool
                               	                        }
                               	FailureMessage: string
                               	// FailedAt is the path to the txn that failed inside of this group
                               	FailedAt: TxnPath []uint64
                              },
		WouldSucceed: true,
}
```
Is this path really just the `FailedAt: TxnPath` inside of **group 0**?

Reply:
And this provokes an uncomfortable question. Should we be future proof and support more than 1 txn group? I have very mixed feelings about it.

Reply:
> And this provokes an uncomfortable question. Should we be future proof and support more than 1 txn group? I have very mixed feelings about it.

This is a good point, I think we should future-proof it. i.e. `the simulation should report a failure at group "0", path "1" with message "transaction already in ledger"`

But just curious why you had mixed feelings about it?

Reply:
"Future proofing" is usually another name for "over-engineering". However, if we know for sure that `/simulate` will support multiple transactions in the near/medium term, then it's not over-engineering, but simply prudent.

@bbroder-algo - any thoughts?

Reply:
I think it will need to support multiple groups eventually - I'm for it

Reply:
I pushed [dcc0b66](https://github.com/algorand/algorand-sdk-testing/pull/265/commits/dcc0b66ec2b133434b8be3c65fa13d01164aa966) which adds the group num in the step.

Reply:
looks good